RealmSwift/Tests/PerformanceTests.swift

```diff
@@ -435,6 +439,92 @@ class SwiftPerformanceTests: TestCase {
         }
     }
 
+    func deleteServerFiles() {
+        try! FileManager.default.removeItem(at: URL(fileURLWithPath: testDir, isDirectory: true).deletingLastPathComponent().appendingPathComponent("mongodb-realm"))
+        App.resetAppCache()
+    }
+
+    func testSyncRealmCacheLookup() {
+        var config = RLMDummyUser().configuration(partitionValue: "")
+        config.objectTypes = []
+        let realm = try! Realm(configuration: config)
+
+        measure {
+            for _ in 0..<1250 {
+                autoreleasepool {
+                    _ = try! Realm(configuration: config)
+                }
+            }
+        }
+        realm.invalidate()
+        deleteServerFiles()
+    }
+
+    func testSyncRealmCreationCached() {
+        var config = RLMDummyUser().configuration(partitionValue: "")
+        config.objectTypes = []
+        var realm: Realm!
+        dispatchSyncNewThread {
+            realm = try! Realm(configuration: config)
+        }
+
+        measure {
+            for _ in 0..<1250 {
+                autoreleasepool {
+                    _ = try! Realm(configuration: config)
+                }
+            }
+        }
+        _ = realm.configuration
```

Comment:
Swift (sometimes) drops references to objects after the last time they are used rather than at the end of the scope, so without this the realm may be deallocated before the actual test runs.